lib/Zonemaster/Backend/Config.pm

```diff
@@ -29,29 +30,152 @@ Readonly my @SIG_NAME => split ' ', $Config{sig_name};
+    $obj->{_MYSQL_password}                                      = $ini->val( 'MYSQL',      'password',                                 undef );
+    $obj->{_MYSQL_database}                                      = $ini->val( 'MYSQL',      'database',                                 undef );
+    $obj->{_POSTGRESQL_host}                                     = $ini->val( 'POSTGRESQL', 'host',                                     undef );
+    $obj->{_POSTGRESQL_user}                                     = $ini->val( 'POSTGRESQL', 'user',                                     undef );
+    $obj->{_POSTGRESQL_password}                                 = $ini->val( 'POSTGRESQL', 'password',                                 undef );
+    $obj->{_POSTGRESQL_database}                                 = $ini->val( 'POSTGRESQL', 'database',                                 undef );
+    $obj->{_SQLITE_database_file}                                = $ini->val( 'SQLITE',     'database_file',                            undef );
+    $obj->{_ZONEMASTER_max_zonemaster_execution_time}            = $ini->val( 'ZONEMASTER', 'max_zonemaster_execution_time',            '600' );
+    $obj->{_ZONEMASTER_maximal_number_of_retries}                = $ini->val( 'ZONEMASTER', 'maximal_number_of_retries',                '0' );
+    $obj->{_ZONEMASTER_number_of_processes_for_frontend_testing} = $ini->val( 'ZONEMASTER', 'number_of_processes_for_frontend_testing', undef );
+    $obj->{_ZONEMASTER_number_of_processes_for_batch_testing}    = $ini->val( 'ZONEMASTER', 'number_of_processes_for_batch_testing',    undef );
+    $obj->{_ZONEMASTER_lock_on_queue}                            = $ini->val( 'ZONEMASTER', 'lock_on_queue',                            undef );
+    $obj->{_ZONEMASTER_age_reuse_previous_test}                  = $ini->val( 'ZONEMASTER', 'age_reuse_previous_test',                  '600' );
+
+    $obj->{_LANGUAGE_locale} = {};
+    for my $locale_tag ( split /\s+/, $ini->val( 'LANGUAGE', 'locale' ) || 'en_US' ) {
+        $locale_tag =~ /^[a-z]{2}_[A-Z]{2}$/
+          or die "Illegal locale tag in LANGUAGE.locale: $locale_tag\n";
+
+        !exists $obj->{_LANGUAGE_locale}{$locale_tag}
+          or die "Repeated locale tag in LANGUAGE.locale: $locale_tag\n";
+
+        $obj->{_LANGUAGE_locale}{$locale_tag} = 1;
+    }
 
-    $self->{cfg} = Config::IniFiles->new( -file => $path );
-    die "UNABLE TO LOAD $path ERRORS:[".join('; ', @Config::IniFiles::errors)."] \n" unless ( $self->{cfg} );
-    bless( $self, $class );
+    $obj->{_public_profiles} = {
+        default => '',
+    };
+    for my $name ( $ini->Parameters( 'PUBLIC PROFILES' ) ) {
+        $obj->{_public_profiles}{lc $name} = $ini->val( 'PUBLIC PROFILES', $name );
+    }
+    $obj->{_private_profiles} = {};
+    for my $name ( $ini->Parameters( 'PRIVATE PROFILES' ) ) {
+        $obj->{_private_profiles}{lc $name} = $ini->val( 'PRIVATE PROFILES', $name );
+    }
```

Comment:
Profile names must match the specification in [API doc] and the profile name must point at a profile file in the file system. I suggest the following tests for profile names:
* Then name must meet the specification.
* The file must exists and being readable by running process.
* The file must be JSON data.

Checking that the file is a valid profile file is too complexed.

[API doc]: https://github.com/zonemaster/zonemaster-backend/blob/master/docs/API.md#profile-name

Reply:
My intention is to limit the scope of this PR by not adding any new validations. In later pull requests I will address all the validations.

While we're on the subject though, the validation I have in mind for filesystem paths is to let  Z::B::Config validate that the value is an absolute path and be satisfied with that. After successfully parsing the configuration but before the daemons fork themselves, the daemons should connect to the database and load all the profile files. This will give us the desired behavior.

Reply:
Ok.